docs/docs/noir/standard_library/bigint.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Big Integers
+description: How to use big integers from Noir standard library
+keywords:
+  [
+    Big Integer,
+    Noir programming language,
+    Noir libraries,
+  ]
+---
+
+The BigInt module in the standard library exposes some class of integers which do not fit (well) into a Noir native field. It implements modulo arithmetic, modulo a 'big' prime number.
+
+:::note
+
+The module can currently be considered as `Field`s with fixed modulo sizes used by a set of elliptic curves, in addition to just the native curve. [More work](https://github.com/noir-lang/noir/issues/510) is needed to achieve arbitrarily sized big integers.
+
+:::
+
+Currently 6 classes of integers (i.e 'big' prime numbers) are available in the module, namely:
+
+- BN254 Fq: Bn254Fq
+- BN254 Fr: Bn254Fr
+- Secp256k1 Fq: Secpk1Fq
+- Secp256k1 Fr: Secpk1Fr
+- Secp256r1 Fr: Secpr1Fr
+- Secp256r1 Fq: Secpr1Fq
+
+Where XXX Fq and XXX Fr denote respectively the order of the base and scalar field of the (usual) elliptic curve XXX.
+For instance the big integer 'Secpk1Fq' in the standard library refers to integers modulo $2^{256}-2^{32}-977$.
+
+Feel free to explore the source code for the other primes:
+
+#include_code curve_order_base noir_stdlib/src/bigint.nr rust
+
+## Example usage
+
+A common use-case is when constructing a big integer from its bytes representation, and performing arithmetic operations on it:
+
+```rust
+use dep::std::{bigint::Secpk1Fq, println};
+
+fn main() {
+ let a = Secpk1Fq::from_le_bytes([x, y, 0, 45, 2]);
+ let b = Secpk1Fq::from_le_bytes([y, x, 9]);
+ let c = (a + b)*b/a;
+ let d = c.to_le_bytes();
+ println(d[0]);
+}
+```
```

Comment:
Can we put this into a snippet to enforce correctness?

Reply:
The println import is valid, I tried this before adding it in the doc, however I did not copy properly because I missed the parameters to main().

Reply:
Ah, yeah it is valid. My mistake.